prinz_pmml/src/main/scala/pl/touk/nussknacker/prinz/pmml/model/PMMLModel.scala

```diff
@@ -1,19 +1,31 @@
 package pl.touk.nussknacker.prinz.pmml.model
 
-import org.jpmml.evaluator.Evaluator
-import org.jpmml.evaluator.LoadingModelEvaluatorBuilder
-import pl.touk.nussknacker.prinz.model.{Model, ModelInstance, ModelName, ModelVersion}
+import org.jpmml.evaluator.{Evaluator, LoadingModelEvaluatorBuilder, PMMLException}
+import pl.touk.nussknacker.prinz.model.{Model, ModelInstance, ModelName, ModelNameNotFoundException, ModelNotValidException, ModelRunException, ModelVersion}
 
-import java.io.{File, InputStream}
+import java.io.InputStream
 
 case class PMMLModel(inputStream: InputStream) extends Model {
-  //TODO we need evaluator in PMMLModel to get the signature
-  //PMMLModelInstance should use this evaluator for scoring
-  val evaluator: Evaluator = new LoadingModelEvaluatorBuilder().load(inputStream).build()
+  var evaluatorBuilder = new LoadingModelEvaluatorBuilder().load(inputStream)
+  var optionalModelName = Option(evaluatorBuilder.getModel().getModelName())
+  val evaluator: Evaluator = evaluatorBuilder.build()
 
-  override def getName: ModelName = ???
+  if (optionalModelName.isEmpty) {
+    throw new ModelNameNotFoundException()
+  }
+  // TODO: Check (on repository level?) if model with such name exists
```

Comment:
imo the repository will be resposible for this check so this comment here is not needed